src/components/ListItem.jsx

```diff
@@ -1,5 +1,70 @@
 import './ListItem.css';
+import { updateItem } from '../api';
+import { useState, useEffect } from 'react';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ item, listPath }) {
+	const { id, name, dateLastPurchased } = item;
+
+	const currentDate = new Date();
+	const lastPurchasedDate = dateLastPurchased
+		? new Date(dateLastPurchased.seconds * 1000)
+		: 0;
+	const timeDiff = currentDate - lastPurchasedDate; //milliseconds
+	const oneDayInMilliseconds = 24 * 60 * 60 * 1000;
+	const [isChecked, setIsChecked] = useState(false);
+
+	useEffect(() => {
+		if (timeDiff < oneDayInMilliseconds) {
+			setIsChecked(true);
+		}
+
+		const timeLeft = oneDayInMilliseconds - timeDiff;
+		const timeoutId = setTimeout(() => {
+			setIsChecked(false);
+		}, timeLeft);
+
+		return () => clearTimeout(timeoutId);
+	}, [isChecked, timeDiff, oneDayInMilliseconds]);
+
+	async function handleCheck(e) {
+		const timeLeftInMilliseconds = oneDayInMilliseconds - timeDiff;
+		if (isChecked && timeLeftInMilliseconds > 0) {
+			const timeLeftInHours = Math.floor(
+				timeLeftInMilliseconds / (1000 * 60 * 60),
+			);
+			const timeLeftInMinutes = Math.floor(
+				(timeLeftInMilliseconds / (1000 * 60)) % 60,
+			);
+			const alertMessage =
+				timeLeftInHours <= 0
+					? `You can set this item to be purchased again in ${timeLeftInMinutes} minutes`
+					: `You can set this item to be purchased again in ${timeLeftInHours} hours and ${timeLeftInMinutes} minutes`;
+			alert(alertMessage);
+		} else {
+			setIsChecked(e.target.checked);
+			if (e.target.checked) {
+				const result = await updateItem(listPath, id);
+				if (result.success) {
+					alert('Item purchased');
+				} else {
+					alert('Error: item not updated');
+				}
+			}
+		}
+	}
+
+	return (
+		<div>
```

Comment:
Since the `listItem`s are being used in an `<ul>`, sticking with `<li>` for its contents is considered best practice with regards to semantic HTML and screen reader experiences.  Changing the `listItem`'s parent `div` to a `li` shouldn't affect the funtionality of the `label` and `input`. 

Unless @mxmason thinks otherwise!

Reply:
thanks for catching that, just made the change!